Fails to run when plugins directory is a docker mounted volume

Because in docker when a directory is marked as a volume, it gets mounted into the container, which means it can't actually be deleted.

```
$ rm -rf /usr/share/jenkins/ref/plugins
rm: cannot remove '/usr/share/jenkins/ref/plugins': Device or resource busy
```

```
$ java -jar /usr/lib/jenkins-plugin-manager.jar -w /usr/share/jenkins/jenkins.war --verbose -f /var/jenkins_home/plugins.txt
File containing list of plugins to be downloaded: /var/jenkins_home/plugins.txt
Reading in plugins from /var/jenkins_home/plugins.txt

No directory to download plugins entered. Will use default of /usr/share/jenkins/ref/plugins
Using update center https://updates.jenkins.io/update-center.json from JENKINS_UC environment variable
Using experimental update center https://updates.jenkins.io/experimental/update-center.json from JENKINS_UC_EXPERIMENTAL environment variable
Using incrementals mirror https://repo.jenkins-ci.org/incrementals from JENKINS_INCREMENTALS_REPO_MIRROR environment variable
No CLI option or environment variable set for plugin info, using default of https://updates.jenkins.io/plugin-versions.json
Will use war file: /usr/share/jenkins/jenkins.war
java.io.UncheckedIOException: Unable to delete: /usr/share/jenkins/ref/plugins
        at io.jenkins.tools.pluginmanager.impl.PluginManager.start(PluginManager.java:129)
        at io.jenkins.tools.pluginmanager.impl.PluginManager.start(PluginManager.java:113)
        at io.jenkins.tools.pluginmanager.cli.Main.main(Main.java:63)
Caused by: java.io.IOException: Unable to delete directory /usr/share/jenkins/ref/plugins.
        at org.apache.commons.io.FileUtils.deleteDirectory(FileUtils.java:1212)
        at io.jenkins.tools.pluginmanager.impl.PluginManager.start(PluginManager.java:127)
        ... 2 more
Unable to delete: /usr/share/jenkins/ref/plugins
```

I think the fix is to delete the contents of the directory, and not the directory itself but having trouble figuring out a clean solution for that. or maybe a flag to make it not delete the directory which would help #108

--- a/plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/impl/PluginManager.java
+++ b/plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/impl/PluginManager.java
@@ -127,11 +127,7 @@ public class PluginManager {
                 File[] toBeDeleted = refDir.listFiles();
                 if (toBeDeleted != null) {
                     for (File deletableFile : toBeDeleted) {
-                        if (deletableFile.isDirectory()) {
-                            FileUtils.deleteDirectory(deletableFile);
-                        } else {
-                            deletableFile.delete();
-                        }
+                        FileUtils.forceDelete(deletableFile);
                     }
                 }
             } catch (IOException e) {

--- a/plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/impl/PluginManager.java
+++ b/plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/impl/PluginManager.java
@@ -125,11 +125,13 @@ public class PluginManager {
         if (refDir.exists()) {
             try {
                 File[] toBeDeleted = refDir.listFiles();
-                for (File deletableFile : toBeDeleted) {
-                    if (deletableFile.isDirectory()) {
-                        FileUtils.deleteDirectory(deletableFile);
-                    } else {
-                        deletableFile.delete();
+                if (toBeDeleted != null) {
+                    for (File deletableFile : toBeDeleted) {
+                        if (deletableFile.isDirectory()) {
+                            FileUtils.deleteDirectory(deletableFile);
+                        } else {
+                            deletableFile.delete();
+                        }
                     }
                 }
             } catch (IOException e) {

--- a/plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/impl/PluginManager.java
+++ b/plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/impl/PluginManager.java
@@ -124,7 +124,14 @@ public class PluginManager {
     public void start(boolean downloadUc) {
         if (refDir.exists()) {
             try {
-                FileUtils.deleteDirectory(refDir);
+                File[] toBeDeleted = refDir.listFiles();
+                for (File deletableFile : toBeDeleted) {
+                    if (deletableFile.isDirectory()) {
+                        FileUtils.deleteDirectory(deletableFile);
+                    } else {
+                        deletableFile.delete();
+                    }
+                }
             } catch (IOException e) {
                 throw new UncheckedIOException("Unable to delete: " + refDir.getAbsolutePath(), e);
             }

